shell/common/pointer_data_dispatcher.h

```diff
@@ -0,0 +1,71 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef POINTER_DATA_DISPATCHER_H_
+#define POINTER_DATA_DISPATCHER_H_
+
+#include "flutter/runtime/runtime_controller.h"
+#include "flutter/shell/common/animator.h"
+
+namespace flutter {
+
+class PointerDataDispatcher {
+ public:
+  virtual void DispatchPacket(std::unique_ptr<PointerDataPacket> packet,
+                              uint64_t trace_flow_id) = 0;
+  virtual void OnRender() = 0;
+  virtual ~PointerDataDispatcher() {}
```

Comment:
Making destructors OOL can lead to reduction in binary size because all the steps needed to destroy an object will not end up in each TU that destroys an object. Please default the same in the .cc.